js/ui/html_editor/modules/toolbar.js

```diff
@@ -503,20 +503,29 @@ if(Quill) {
                     continue;
                 }
 
-                this._markActiveFormatWidget(formatName, formatWidget, formats);
+                this._markActiveFormatWidget(formatName, formatWidget, formats, selection);
             }
 
             this._toggleClearFormatting(hasFormats || selection.length > 1);
         }
 
-        _markActiveFormatWidget(name, widget, formats) {
+        _markActiveFormatWidget(name, widget, formats, selection) {
             if(this._isColorFormat(name)) {
                 this._updateColorWidget(name, formats[name]);
             }
 
             if('value' in widget.option()) {
                 this._setValueSilent(widget, formats[name]);
             } else {
+                if(selection?.length === 0) { // T1157840
+                    const nextPositionFormat = this.quill.getFormat({
+                        length: 0,
+                        index: selection.index + 1
+                    });
+                    if(!nextPositionFormat[name]) {
```

Comment:
What if there is other format on the next index?

Reply:
if next position has another format, this case will(or did) processed in next(prev) iteration of [loop](https://github.com/DevExpress/DevExtreme/blob/de6ce8e404be8d629a77ef49cc1d2e6e9948d266/js/ui/html_editor/modules/toolbar.js#L498)
>We will still show an empty link?
yes